src/main/java/hudson/plugins/ec2/EC2Cloud.java

```diff
@@ -510,56 +526,78 @@ private int getPossibleNewSlavesCount(SlaveTemplate template) throws AmazonClien
      * Obtains a slave whose AMI matches the AMI of the given template, and that also has requiredLabel (if requiredLabel is non-null)
      * forceCreateNew specifies that the creation of a new slave is required. Otherwise, an existing matching slave may be re-used
      */
-    private synchronized EC2AbstractSlave getNewOrExistingAvailableSlave(SlaveTemplate template, Label requiredLabel, boolean forceCreateNew) {
+    private EC2AbstractSlave getNewOrExistingAvailableSlave(SlaveTemplate template, Label requiredLabel, boolean forceCreateNew) {
```

Comment:
We need to remove `synchronized` in order to provision nodes in parallel. The consequence is that we might provision more nodes than we want.